magicbot/magicrobot.py

```diff
@@ -456,6 +456,31 @@ def _setup_vars(self, cname, component):
             setattr(component, n, injectable)
             self.logger.debug("-> %s as %s.%s", injectable, cname, n)
         
+        if hasattr(component, '__annotations__'):
+            for n, type_ in component.__annotations__.items():
+                if n.startswith('_') or isinstance(getattr(type(component), n, True), property):
```

Comment:
Uhm. That's not what you're doing at all.

`isinstance(getattr(type(component), n, True), property)` implies `hasattr(type(component), n)`, for obvious reasons.